### PR TITLE
Fix WeightToFee function in MangataAdapter

### PR DESCRIPTION
**Pull request by imstar15**
According Mangata code

https://github.com/mangata-finance/mangata-node/blob/24d4304b982c02ed162f2ff5d94faa907520b3fd/runtime/mangata-rococo/src/constants.rs#L49

If we use MGR for xcmp fee, We should use the following code to calculate the fee by weight.

```
const UNIT = new BN("1000000000000000000");
// ExtrinsicBaseWeight benchmark value: 114756 nano seconds
const extrinsicBaseWeight = WEIGHT_REF_TIME_PER_NANOS.mul(new BN(114756));
const feePerSecond = WEIGHT_REF_TIME_PER_SECOND.mul(
  UNIT.div(extrinsicBaseWeight),
);
const refTimeFee = weight.refTime
  .mul(feePerSecond)
  .div(WEIGHT_REF_TIME_PER_SECOND);
const proofSizeFee = weight.proofSize
  .mul(feePerSecond)
  .div(WEIGHT_PROOF_SIZE_PER_MB);
return refTimeFee.add(proofSizeFee);
```

### CODE REVIEW

**Review by chrisli30 — DISMISSED** [review]
The code looks generally good to me.

**Review by chrisli30 — APPROVED** [review]
LGTM, but I will find a better name for the `x2Params` since it’s not accurate enough.

**Review by chrisli30 — CHANGES_REQUESTED** [review]
Let’s revert the line break changes. Please see what is the value of your eslint `max-len`

File: packages/adapter/src/chains/oak.ts [hunk 1]
```diff
@@ -422,13 +563,20 @@ export class OakAdapter extends ChainAdapter {
    * @param keyringPair
    * @returns
    */
-  async bondMore(collator: HexString, amount: BN, keyringPair: KeyringPair): Promise<SendExtrinsicResult> {
+  async bondMore(
+    collator: HexString,
+    amount: BN,
+    keyringPair: KeyringPair,
+  ): Promise<SendExtrinsicResult> {
```
[the remark this answers is not in the document]
Comment (imstar15) [reply packages/adapter/src/chains/oak.ts:570]: my eslint config:

```
"max-len": [
      0,
      100,
      2,
      {
        "ignoreUrls": true,
        "ignoreComments": false,
        "ignoreRegExpLiterals": true,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": true
      }
```

[the remark this answers is not in the document]
Comment (imstar15) [reply packages/adapter/src/chains/oak.ts:570]: Replace `collator:·HexString,·amount:·BN,·keyringPair:·KeyringPair` with `⏎····collator:·HexString,⏎····amount:·BN,⏎····keyringPair:·KeyringPair,⏎··` eslint([prettier/prettier](https://github.com/prettier/eslint-plugin-prettier#options))

[the remark this answers is not in the document]
Comment (chrisli30) [reply packages/adapter/src/chains/oak.ts:570]: Add the same content to root folder. Let’s try to use the set rules.
https://github.com/OAK-Foundation/automation-hub/blob/staging/.prettierrc

File: packages/config/src/tokens/types/Token.ts [hunk 2]
```diff
@@ -2,60 +2,66 @@
...
+  decimals: number;
+  network: string;
+  parachainId: number;
+  x2Params?: any;
 }
 
 // Function to construct the 'location' object based on the parameters.
-function constructLocation(parachainId: number, palletInstance?: number): Location {
-	let interior = palletInstance ? { X2: [{ Parachain: parachainId }, { PalletInstance: palletInstance }] } : { X1: { Parachain: parachainId } };
+function constructLocation(parachainId: number, x2Params?: any): Location {
```
Comment (chrisli30) [inline packages/config/src/tokens/types/Token.ts:48]: Let’s rename the `x2Params` to `interior`.

**Review by chrisli30 — DISMISSED** [review]

File: packages/config/src/tokens/types/Token.ts (hunk 2 above)
Comment (imstar15) [reply packages/config/src/tokens/types/Token.ts]: Fixed

**Review by chrisli30 — APPROVED** [review]
Okay, the changes look good 🆗

**Pull request merged**